ow_lander/scripts/activity_full_digging_traj.py

```diff
@@ -32,6 +32,24 @@ def go_to_Z_coordinate(move_group, x_start, y_start, z_start):
   move_group.clear_pose_targets()
 
 
+def go_to_Z_cartesian(move_group, z_start):
+  """
+  :type move_group: class 'moveit_commander.move_group.MoveGroupCommander'
+  :type z_start: float
+  """
+  wpose = move_group.get_current_pose().pose
+  wpose.position.z = z_start
```

Comment:
To compensate the shift we could have a 3D cartesian motion. Now it's 1D because along Z. We could have a motion along X, Y, Z to make sure that the final coordinates of the end effector are x_start, y_start, z_start.

Reply:
I was trying this actually. I think it is sufficient to set only the X, Z since the hand retreats a bit backward a bit on the X direction.

Reply:
> I was trying this actually. I think it is sufficient to set only the X, Z since the hand retreats a bit backward a bit on the X direction.
This is for the specific case of the arm scooping at the center of the workspace. The offset might change when scooping on the sides, gaining a Y component as well.

Reply:
When trying to give the cartesian planner the 3D goal point (x_star, y_star, z_star), the planner doesn't find any trajectory to get there. This is because the cartesian planner uses "pose" (position+orientation) waypoints and I am telling the planner to move the hand to another position while keeping the same orientation throughout the trajectory, which is not possible with our arm configuration. So the solution is to place the hand on the point (x_star, y_star, z_star) using _go_to_Z_coordinate_, before calling _go_to_Z_cartesian_. I have implemented this on a copy of this branch called OCEANWATER-518_Switch_to_using_effort_controllers_JointTrajectoryController_DAMIANA. In this branch I also: 1) multiplied by 10 the P gains for all the joint controllers, 2) set RRTstar as default planner (apparently default was still RRTconnect). I am not creating a ticket/PR because it addresses the same problem that your ticket/PR addresses. If you like, you can merge mine into yours.

Reply:
You can create a separate PR that would merge into this branch (no jira ticket needed).
This what I was trying to deal with .. the issue when adding x, y to the goal of go_to_Z_cartesian is that limbs move_group_commander fails to generate a proper solution. There are cases where it succeeds and on other occasions it fails (for the exact same goal). I am trying to understand the problem better before settling on a particular a solution.

Reply:
Ultimately, we need to find something reliable by tomorrow.

Reply:
I don't seem to see the branch .. did you forget to publish it?

Reply:
Sorry, the branch is called "bugfix/rrtconnect_unable_to_sample_any_valid_states_for_goal_tree_DAMIANA". Also, I can explain why it fails.